src/jit/lir.cpp

```diff
@@ -373,10 +380,52 @@ GenTree*& LIR::Range::LastNode() const
 //
 // Return Value:
 //
+// REVIEW: should this be named AssertIsValid, so we can convert these
+// checks into asserts that will fire at the exact place a corrupt
+// component is found, so it is easier to see what is asserting instead
+// of a generic "assert(IsValid())".
 bool LIR::Range::IsValid() const
```

Comment:
This is really only intended as a lightweight usability check, not an IR correctness check. The added checks should be moved to `CheckLIR`.